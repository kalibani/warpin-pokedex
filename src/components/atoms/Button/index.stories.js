// Stories Button Component
// --------------------------------------------------------
import React, { Fragment } from 'react';
import { storiesOf } from '@storybook/react';
import Button from './index';
import Notes from './readme.md';

storiesOf('Atoms', module).add('Button', () => (
<Button />
), {
info: { inline: true, header: false, text: Notes },
}
);
