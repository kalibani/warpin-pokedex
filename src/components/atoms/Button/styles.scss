/* ==========================================================================
Styles for Button component
========================================================================== */

// import mixin and variables
@import "../../../assets/styles/variables";

// Start styles Button component
.a-button {

}
