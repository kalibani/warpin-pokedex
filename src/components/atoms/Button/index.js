/* eslint-disable react/button-has-type */
// Button Component
// --------------------------------------------------------

import React from 'react';
import PropTypes from 'prop-types';
import classname from 'classnames';
import './styles.scss';

const Button = ({
  className,
  style,
  variant,
  family,
  tabIndex,
  onClick,
  disabled,
  full,
  type,
  children
}) => {
  const classNames = classname('a-button', className, variant, family, {
    'full-content': full
  });
  return (
    <>
      <button
        className={classNames}
        type={type}
        onClick={onClick}
        tabIndex={tabIndex}
        style={style}
        disabled={disabled}
      >
        {children}
      </button>
    </>
  );
};

Button.propTypes = {
  className: PropTypes.string,
  style: PropTypes.object,
  variant: PropTypes.string,
  family: PropTypes.string,
  tabIndex: PropTypes.string,
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
  full: PropTypes.bool,
  type: PropTypes.string,
  children: PropTypes.node
};

Button.defaultProps = {
  className: '',
  style: {},
  variant: '',
  family: '',
  tabIndex: '0',
  onClick: () => {},
  disabled: false,
  full: false,
  children: '',
  type: 'button'
};

export default Button;
