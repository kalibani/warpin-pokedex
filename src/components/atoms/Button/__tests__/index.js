// Test Button Component
// --------------------------------------------------------

import React from 'react';
import Button from '../index';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

describe('<Button />', () => {
  describe('onClick()', () => {
    test('successfully calls the onClick handler and match the snapshot', () => {
      const mockOnClick = jest.fn();
      const wrapper = shallow(
        <Button onClick={mockOnClick}>Click Me</Button>
      );

      wrapper.find('button').simulate('click');
      expect(mockOnClick.mock.calls.length).toEqual(1);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});


