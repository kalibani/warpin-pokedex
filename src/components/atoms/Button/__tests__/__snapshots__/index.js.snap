// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> onClick() successfully calls the onClick handler and match the snapshot 1`] = `
<Fragment>
  <button
    className="a-button"
    disabled={false}
    onClick={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    style={Object {}}
    tabIndex="0"
    type="button"
  >
    Click Me
  </button>
</Fragment>
`;
